### DISCUSSION TITLE
Move mime_classifier to a separate crate

### DISCUSSION THREAD

**Discussion by jedenastka** [servo/servo#33415] [Ideas]
**Describe the new feature:**
Move `components/net/mime_classifier.rs` to a separate crate, so it can be reused by different projects. I think there is at least some demand for a "browser-grade" media type classifier component, especially when building software that interacts with the web in similar manner to a browser, where using libraries such as [file-format](https://crates.io/crates/file-format) is not ideal.

There is already a [mime_classifier](https://crates.io/crates/mime_classifier) crate published, which uses code from Servo, but it seems abandoned. If reusing that name would be desirable, the current maintainer mentions in the description:

> If you are a member of the Servo and wish to maintain this library yourself, I'd be glad to transfer ownership of the crate: please open a GitHub issue or send me an email.

**Comment by jdm** [comment]
I'm open to this idea. It's code that is rarely touched, so it shouldn't be a problem to extract it into its own repository for simplicity (since we don't publish any crates that live inside servo/servo at this point).

**Comment by demurgos** [comment]
Hello :wave: 
I'm the maintainer of the `mime_classifier` crate. I needed the feature as a standalone crate so I extracted and published it. I'm still relying on it, I just did not have a need to publish a new version since then. As mentioned in the README, I would much prefer if the crate was directly published by Servo. I can help send a PR to move the classifier module into its own crate.

**Comment by cablehead** [comment]
There's some related discussion here:

https://github.com/demurgos/mime_classifier/issues/1#issuecomment-2427526460

Transferring the `mime_classifier` crate as @demurgos  offers to the core Servo team seems ideal. What would it look like to  extract it into its own repository?

**Comment by jdm** [comment]
We discussed the idea in zulip, and there isn't a lot of enthusiasm to take on maintainance of another crate at this point in time. Additionally, the code hasn't been modified since the crate was published, so members of the Servo team don't see any particular urgency for making a decision about it.

**Reply by jdm** [reply]
https://servo.zulipchat.com/#narrow/stream/263398-general/topic/Extracting.20mime.20classifier.20implementation.20into.20a.20crate
